src/main/java/dev/socketmods/socketnukes/block/explosive/TNTExplosive.java

```diff
@@ -41,6 +45,27 @@ public TNTExplosive() {
         super(AbstractBlock.Properties.create(Material.TNT));
     }
 
+    /**
+     * Sets the short-lived tile entity data when this block is placed by the BlockEntity.
+     * This is just a hacky way of persisting the data inside the block.
+     * This should use WorldSavedData or something.
+     * @param worldIn The world where the block was placed
+     * @param pos The position the block was placed at
+     * @param state The new blockstate of this block.
+     * @param placer The entity that placed the block
+     * @param stack The stack the entity was holding when it was placed.
+     */
+    @Override
+    public void onBlockPlacedBy(World worldIn, BlockPos pos, BlockState state, @Nullable LivingEntity placer, ItemStack stack) {
+        super.onBlockPlacedBy(worldIn, pos, state, placer, stack);
+
+        if(stack.getItem() == SNRegistry.GENERIC_EXPLOSIVE_ITEM.get()
+                && worldIn.getTileEntity(pos) != null && worldIn.getTileEntity(pos) instanceof ExplosiveTileEntity) {
+            ResourceLocation config = new ResourceLocation(stack.getOrCreateChildTag(SocketNukes.MODID).getString("explosion"));
+            ((ExplosiveTileEntity) Objects.requireNonNull(worldIn.getTileEntity(pos))).setConfiguration(config);
```

Comment:
I don't believe the isLoaded checks are necessary here, this function should only ever be called in loaded chunks to my knowledge. I'm going to leave the check out for the moment, but this may be worth a second look.